kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMDeviceStateTest.java

```diff
@@ -0,0 +1,181 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ ******************************************************************************/
+package org.eclipse.kura.nm.configuration;
+
+import static org.junit.Assert.assertEquals;
+
+import org.eclipse.kura.KuraException;
+import org.freedesktop.dbus.types.UInt32;
+import org.junit.Test;
+
+public class NMDeviceStateTest {
+	
+	NMDeviceState state;
+	
+    @Test
+    public void shouldReturnCorrectStateTypeZero() throws InterruptedException, KuraException {
```

Comment:
I would also use the more meaningful representation for naming the test. For example in this case instead of using the *source data* (i.e. `UInt32(0)`) use the *target data* (i.e. `STATE_UNKNOWN`).

Furthermore I would use the `conversionWorksFor*` naming convention.

All in all something like the following:

```suggestion
    public void shouldReturnCorrectStateTypeZero() throws InterruptedException, KuraException {
```

```suggestion
    public void conversionWorksForStateUnknown() throws InterruptedException, KuraException {
```